server/test/utils.ts

```diff
@@ -0,0 +1,94 @@
+import { rand, randEmail, randNumber, randUserName, randQuote, randBrand } from '@ngneat/falso';
+
+import { User } from 'server/models';
+
+// user interface
+interface UserObject {
+  email?: string;
+  discord_user_id?: string;
+  discord_name?: string;
+  bio?: string;
+  twitter_username?: string;
+  linkedin_url?: string;
+  github_username?: string;
+  website?: string;
+}
+
+// https://stackoverflow.com/questions/39494689/is-it-possible-to-restrict-number-to-a-certain-range/39495173#39495173
+// Tim's fault for telling me this is possible
+type Enumerate<N extends number, Acc extends number[] = []> = Acc['length'] extends N
+  ? Acc[number]
+  : Enumerate<N, [...Acc, Acc['length']]>;
+
+type IntRange<F extends number, T extends number> = Exclude<Enumerate<T>, Enumerate<F>>;
+
+// random
+// probability should only be 1 - 100
+function randomEmptyChance(probability: IntRange<0, 100>, preferredResult: string) {
+  // out of a 100
+  const randomNum = Math.random() * 100;
+  return randomNum < probability ? '' : preferredResult;
+}
+
+// example
+
+function randDiscordUserName() {
+  return `${randUserName({})}#${String(
+    randNumber({
+      min: 0,
+      max: 9999,
+    })
+  ).padStart(4, '0')}`;
+}
+
+function randUserNameHandle() {
+  return `@${randUserName({})}`;
+}
+
+function randLinkedInURL() {
+  return `https://www.linkedin.com/in/${randUserName()}/`;
+}
+
+function randWebsite() {
+  const scheme = rand(['https', 'http']);
+  const wwwPart = rand(['www.', '']);
+  const regex = new RegExp('[\\W]', 'g');
+  const host = randBrand().replace(regex, '');
+  const tld = rand(['com', 'gg', 'dev', 'org', 'net', 'me', 'io', 'us', 'info']);
+
+  return `${scheme}://${wwwPart}${host}.${tld}`;
+}
```

Comment:
we can use falso for this: https://ngneat.github.io/falso/docs/internet#randurl

Reply:
I'm cool with randurl and didn't see it before but the way it's implemented doesn't give a chance for 'www' to show up.

Will that create a blind spot for testing logic and UI?

from https://github.com/ngneat/falso/blob/main/packages/falso/src/lib/url.ts
```ts
 return `${fake(['http', 'https'])}://${randWord()}.${randDomainSuffix()}`;
```

Reply:
nope! `www` is not required for URLs anyway